### ISSUE TITLE
Gmail: sometimes gets new mail, sometimes misses it indefinitely

### ISSUE THREAD

**Issue by jdnixx** [ikvk/imap_tools#153]
I'm trying to automate retrieving verification code emails from Huobi (crypto exchange) by connecting to my gmail acct. I've got a while loop searching for new (unread) messages which terminates when it finds one and it parses it. It calls this function in the body of the loop to grab mail every 1 second:

```python
def get_messages(self):
        self.messages = self.mailbox.fetch(
            criteria=AND(
                from_='huobi',  # substrings matches work
                subject='Security Verification',
                date_gte=datetime.date.today(),  # SINCE 15-Mar-2000
                seen=False),
            mark_seen=True,
            bulk=True
        )
```

When I debug it, I can see that the `mailbox` (which is a: `MailBox('imap.gmail.com').login(username, password)`) successfully authenticates every time no problem. However. Sometimes when I run the program, the code email gets sent from Huobi and it gets retrieved with imap-tools within one or two iterations of the while loop (so 1-2seconds); other times, the loop runs infinitely, never finding the new message, even after I go to another tab with Gmail open and manually refresh it and see the unread message right there. And this is all despite authenticating through imap-tools no problem. It's just not finding the new message sometimes.

Is there any way to figure out what could be causing it to happen, especially as intermittently as it is? I can't see anything different even when debugging, between the times it gets the mail successfully and when it fails to get it + just runs the loop indefinitely.

**Comment by ikvk** [comment]
Try it 
https://github.com/ikvk/imap_tools/issues/142

You may think about IDLE implementation
https://github.com/ikvk/imap_tools/issues/93

**Comment by jdnixx** [comment]
Thanks for that, I didn't know someone else noticed the same type of behavior. Indeed I ended up fixing this a couple days ago after realizing that I was instantiating a mailbox object once (with the .login() function) and that remained for the life of the class, so it might not've been retrieving new mail. I moved the mailbox into a loop I was using for parsing the incoming mail and made it do a fresh login over & over, once per second, which worked great.

Coming back to this though because I was just testing now and got this:
`imap_tools.errors.MailboxLoginError: Response status "OK" expected, but "NO" received. Data: [b'[ALERT] Too many simultaneous connections. (Failure)']`

which i was afraid of - looks like gmail throttled me a bit. I'll just increase the timeout between loop iterations to 3s or something but not sure what else could be done
